lcls_tools/common/data_analysis/fit/methods.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from scipy.stats import norm, gamma
+from lcls_tools.common.data_analysis.fit.method_base import MethodBase
+
+
+class GaussianModel(MethodBase):
+    """
+    GaussianModel Class that finds initial parameter values for gaussian distribution
+    and builds probability density functions for the likelyhood a parameter
+    to be that value based on those initial parameter values.
+    Passing this class the variable profile_data automatically updates
+    the initial values and and probability density functions to match that data.
+    """
+
+    param_names: list = ["mean", "sigma", "amplitude", "offset"]
+    param_bounds: np.ndarray = np.array(
+        [
+            [0.01, 1.0],
+            [0.01, 5.0],
+            [0.01, 1.0],
+            [0.01, 1.0],
+        ]
+    )
+
+    def find_init_values(self) -> dict:
+        """Fit data without optimization, return values."""
+
+        data = self._profile_data
+        x = np.linspace(0, 1, len(data))
+        # init_fit = norm.pdf(data)
+        offset = data.min()
+        amplitude = data.max() - offset
+
+        weighted_mean = np.average(x, weights=data)
+        weighted_sigma = np.sqrt(np.cov(x, aweights=data))
+
+        self.init_values = {
+            self.param_names[0]: weighted_mean,  # data.mean()
```

Comment:
The text keys are more clear than the numbered indices here. I don't imagine these param names are going to change all that often. Maybe a pydantic structure would be a better way of accomplishing what this does?

Reply:
I agree. We can discuss what that would look like in our meeting.